src/io/config.cpp

```diff
@@ -178,7 +178,12 @@ void GetDeviceType(const std::unordered_map<std::string, std::string>& params, s
     } else if (value == std::string("cuda")) {
       *device_type = "cuda";
     } else if (value == std::string("cuda_exp")) {
-      *device_type = "cuda_exp";
+      Log::Warning(
```

Comment:
I see. Now that we keep `cuda_exp` as an valid option with warning here, maybe it is reasonable to keep `USE_CUDA_EXP` in the make file.

Reply:
I just pushed https://github.com/microsoft/LightGBM/commit/ac7ab774d5152478aa660028c2ee0d866a008daa completely removing remaining uses of `CUDA_EXP`, based on https://github.com/microsoft/LightGBM/pull/5677#discussion_r1089889981